library/kotlin/io/envoyproxy/envoymobile/stats/Stats.kt

```diff
@@ -0,0 +1,13 @@
+package io.envoyproxy.envoymobile
+
+/** Interface for stats */
+interface Stats {
+
+  fun convert(tags: List<Tag>): Map<String, String> {
+    val tagMap = mutableMapOf<String, String>()
+    tags.forEach { it ->
+      tagMap[it.key] = it.value
+    }
+    return tagMap
```

Comment:
This interface is created to host the shared logic to convert list of `Tag` to `Map<String, String>`

Reply:
My Kotlin is a little rusty but I wonder if we could simplify this by using an extension method instead, something like
```kotlin
internal fun List<Tag>.toMap(): Map<String, String> { ... }
```

Reply:
I thought about it, and wasn't sure about the opinion for having kotlin extension here - my reading is different codebase tends to have different preferences on kotlin extension.

Similarly for iOS, the static convert function can be facilitated with an extension too.

Would like to hear @junr03 and @goaway 's thoughts on this.

Reply:
Why do we convert it to a map at all? Isn't it simpler to keep it as a list, given we'll be passing it to the JNI as an array anyways?

Reply:
Reading this conversation makes me realize that Envoy won't treat the list of tags like a set, so if we can ensure consistent ordering (e.g. via an ordered map) we would avoid duplicate stat entries that really has the same set of tags. 

Maybe this should be fixed in Envoy, I'll take a look

Reply:
👍
fwiw, currently tags are passed into envoy as [Stats::StatNameTagVector](https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/include/envoy/stats/tag.h#L27)